connexion/apis/aiohttp_api.py

```diff
@@ -293,13 +293,50 @@ async def get_request(cls, req):
         :rtype: ConnexionRequest
         """
         url = str(req.url)
-        logger.debug('Getting data and status code',
-                     extra={'has_body': req.has_body, 'url': url})
+
+        logger.debug(
+            'Getting data and status code',
+            extra={
+                # has_body | can_read_body report if
+                # body has been read or not
+                # body_exists refers to underlying stream of data
+                'body_exists': req.body_exists,
+                'can_read_body': req.can_read_body,
+                'content_type': req.content_type,
+                'url': url,
+            },
+        )
 
         query = parse_qs(req.rel_url.query_string)
         headers = req.headers
         body = None
-        if req.body_exists:
+
+        # if request is not multipart, `data` will be empty dict
+        # and stream will not be consumed
+        post_data = await req.post()
+
+        # set those up beforehand, they are needed anyway
+        files = {}
+        form = {}
+
+        if post_data:
+            logger.debug('Reading multipart data from request')
+            for k, v in post_data.items():
+                if isinstance(v, web.FileField):
+                    if k in files:
+                        try:
+                            files[k].append(v)
```

Comment:
unless you want to push for #987 being merged first, I'd suggest that put a decent home here explaining that this is nasty workaround or something ;)

Reply:
I indicated in the opening notes that this would replace PR #987 (since it's based on it). I was just trying to finish what you started since you indicate you had moved on to axion.

by "nasty workaround", are you referring to the try/except nature of it?  It's just trying to put files which have the same name into a list.. not sure how that should happen, this isn't a query string.  But hey.

Reply:
well, just look at that, does that look ok to you? figuring that out was one pain point, understanding that will be another one :(

Reply:
and done